src/NZSL/Ast/SanitizeVisitor.cpp

```diff
@@ -331,6 +332,17 @@ namespace nzsl::Ast
 					return HandleIdentifier(identifierData, node.identifiers.front().sourceLocation);
 				}
 			}
+
+			if (identifierData && identifierData->category == IdentifierCategory::External)
+			{
+				identifierData = FindIdentifier(identifierExpr.identifier + node.identifiers.front().identifier);
+				if (identifierData)
+				{
+					auto variableValuePtr = HandleIdentifier(identifierData, node.identifiers.front().sourceLocation);
+					static_cast<VariableValueExpression*>(variableValuePtr.get())->prefix = identifierExpr.identifier;
```

Comment:
I'm not really a fan of this one, and I'm not really sure how to tackle it correctly